src/host/screenInfo.cpp

```diff
@@ -1124,15 +1129,11 @@ void SCREEN_INFORMATION::_InternalSetViewportSize(const til::size* const pcoordS
             // If the new bottom is supposed to be before the final line of the buffer
             // Check to ensure that we don't hide the prompt by collapsing the window.
 
-            // The final valid end position will be the coordinates of
-            // the last character displayed (including any characters
-            // in the input line)
-            til::point coordValidEnd;
-            Selection::Instance().GetValidAreaBoundaries(nullptr, &coordValidEnd);
+            const auto coordValidEnd = _textBuffer->GetCursor().GetPosition();
 
             // If the bottom of the window when adjusted would be
             // above the final line of valid text...
-            if (srNewViewport.bottom + DeltaY < coordValidEnd.y)
+            if (sBottomProposed < coordValidEnd.y)

```

Comment:
The `GetValidAreaBoundaries` will not have been updated to reflect the new prompt position at this point, so we can't use that anymore to determine the viewport position. As an alternative I'm now using the cursor position. That will work a little worse for very long prompts, but at least it works. We can improve this later on.